tests/Aspire.Hosting.MySql.Tests/MySqlFunctionalTests.cs

```diff
@@ -0,0 +1,341 @@
+using Aspire.Hosting.Testing;
+using Aspire.Hosting.Utils;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using MySqlConnector;
+using Polly;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Aspire.Hosting.MySql.Tests;
+
+public class MySqlFunctionalTests(ITestOutputHelper testOutputHelper)
+{
+    [Fact]
+    [RequiresDocker]
+    public async Task VerifyMySqlResource()
+    {
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(5));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(1), ShouldHandle = new PredicateBuilder().Handle<MySqlException>() })
+            .Build();
+
+        var builder = CreateDistributedApplicationBuilder();
+
+        var mySqlDbName = "db1";
+
+        var mysql = builder.AddMySql("mysql").WithEnvironment("MYSQL_DATABASE", mySqlDbName);
+        var db = mysql.AddDatabase(mySqlDbName);
+
+        using var app = builder.Build();
+
+        await app.StartAsync();
+
+        var hb = Host.CreateApplicationBuilder();
+
+        hb.Configuration.AddInMemoryCollection(new Dictionary<string, string?>
+        {
+            [$"ConnectionStrings:{db.Resource.Name}"] = await db.Resource.ConnectionStringExpression.GetValueAsync(default)
+        });
+
+        hb.AddMySqlDataSource(db.Resource.Name);
+
+        using var host = hb.Build();
+
+        await host.StartAsync();
+        await pipeline.ExecuteAsync(async token =>
+        {
+            using var connection = host.Services.GetRequiredService<MySqlConnection>();
+            await connection.OpenAsync(token);
+
+            var command = connection.CreateCommand();
+            command.CommandText = $"SELECT 1";
+            var results = await command.ExecuteReaderAsync(token);
+
+            Assert.True(results.HasRows);
+        }, cts.Token);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    [RequiresDocker]
+    public async Task WithDataShouldPersistStateBetweenUsages(bool useVolume)
+    {
+        var mySqlDbName = "tempdb";
+
+        string? volumeName = null;
+        string? bindMountPath = null;
+
+        var cts = new CancellationTokenSource(TimeSpan.FromMinutes(3));
+        var pipeline = new ResiliencePipelineBuilder()
+            .AddRetry(new() { MaxRetryAttempts = 10, Delay = TimeSpan.FromSeconds(1), ShouldHandle = new PredicateBuilder().Handle<MySqlException>() })
+            .Build();
+
+        try
+        {
+            var builder1 = CreateDistributedApplicationBuilder();
+
+            var password = "p@ssw0rd1";
+
+            var passwordParameter = builder1.AddParameter("pwd");
+            builder1.Configuration["Parameters:pwd"] = password;
+            var mysql1 = builder1.AddMySql("mysql", passwordParameter).WithEnvironment("MYSQL_DATABASE", mySqlDbName);
```

Comment:
Instead of hard-coding a password (and cause more cred scan issues), can we use `ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter` instead?

https://github.com/dotnet/aspire/blob/e38416af996811aa904b262ad21c6532c0d3f446/src/Aspire.Hosting/ParameterResourceBuilderExtensions.cs#L140-L162